src/pages/_app.jsx

```diff
@@ -1,95 +1,80 @@
 
-const WrappedApp = ({ Component, pageProps, req }) => {
+const WrappedApp = ({ Component, pageProps }) => {
+  const { httpError } = pageProps;
   const router = useRouter();
-  const [experimentId, setExperimentId] = useState(undefined);
 
-  console.log(req, pageProps);
+  const { auth } = useSelector((state) => state.networkResources);
+
+  const { experimentId } = router.query;
+  const experimentData = useSelector((state) => (experimentId ? state.experimentSettings.info : {}));
 
-  // Only hydrate pages when experiment ID is loaded.
   useEffect(() => {
-    if (!router.route.includes('experimentId')) {
-      setExperimentId(false);
-    }
+    const { userPoolId, identityPoolId, userPoolClientDetails } = auth;
 
-    if (router.route.includes('experimentId')) {
-      setExperimentId(router.query.experimentId);
+    if (!auth.userPoolId || !auth.identityPoolId) {
+      return;
     }
 
-    setupAmplify(getCurrentEnvironment());
-  }, [router.query.experimentId]);
-
-  const { data: experimentData, error: experimentError } = useSWR(
-    () => (experimentId ? `/v1/experiments/${experimentId}` : null),
-    getFromApiExpectOK,
-  );
+    Amplify.configure({
+      ...configure(userPoolId, identityPoolId, userPoolClientDetails),
+      ssr: true,
+    });
+  }, [auth]);
 
   const mainContent = () => {
-    // If the page is statically rendered (on server), show a loading context.
-    if (!isBrowser) {
-      return (
-        <PreloadContent />
-      );
-    }
-
     // If this is a not found error, show it without the navigation bar.
     if (Component === NotFoundPage) {
       return <Component {...pageProps} />;
     }
 
-    // If the experiment ID does not exist and is not `false` (i.e. not necessary)
-    // wait until the experiment ID is loaded before loading the page.
-    if (!experimentId && experimentId !== false) {
-      return (<PreloadContent />);
-    }
-
-    // If we found the experiment ID, but we haven't yet queried the API for data
-    // about the experiment, wait until that is done.
-    if (experimentId && !experimentData && !experimentError) {
-      return (<PreloadContent />);
-    }
-
     // If there was an error querying the API, display an error state.
-    if (experimentError) {
-      if (experimentError.payload === undefined) {
-        return <Error errorText='Cannot connect to API service.' />;
+    if (httpError) {
+      if (httpError === 404) {
+        return (
+          <NotFoundPage
+            title={'Analysis doesn\'t exist.'}
+            subTitle={'We searched, but we couldn\'t find the analysis you\'re looking for.'}
+            hint='It may have been deleted by the project owner. Go home to see your own priejcts and analyses.'
+          />
+        );
       }
 
-      const { status } = experimentError.payload;
-
-      if (status === 404) {
-        return <NotFoundPage />;
-      }
-
-      return <Error statusCode={status} />;
+      return <Error statusCode={httpError} />;
```

Comment:
Similarly, all errors are thrown on server-side and passed to the app in a `httpError` prop.